src/usePlaidLink.ts

```diff
@@ -39,6 +39,12 @@ export const usePlaidLink = (options: PlaidLinkOptions) => {
       return;
     }
 
+    // if an old plaid instance exists, destroy it before
+    // creating a new one
+    if (plaid != null) {
+      plaid.destroy();
```

Comment:
note: `destroy()` will error if the plaid instance is visible - not sure if this is the case here but might be worth calling `plaid.exit({ force: 1 })` before this

Reply:
This `plaid.destroy` function here is actually a wrapper around the internal destroy method.

This wrapper actually does call `exit({ force: true })` here https://github.com/plaid/react-plaid-link/pull/95/files#diff-aec11f5a5e50b0370cb24fa70752336bR62 :)

Reply:
might be needed, see my error below

Reply:
hmm suggestion doesn't seem to fix it

Reply:
also needs to be added here and then it works https://github.com/atomkirk/react-plaid-link/commit/8167301fa017163b23dca1775c507eb09720a339

Reply:
this seems to break it: `setTimeout(cleanup, 1000);`

I think this might be an issue that needs to be resolved in the initializer

Reply:
The error could be thrown if we reopen Link within the 1 second where Link is closed but `cleanup` here has not been called.

Is that what's happening to you?

Reply:
yes I think so. I've been trying to figure this out for hours and can't find any way to safely and immediately destroy it before recreating it in a way that always works.

So, I'm not sure how to fix. I suspect the best way is to update the initialize script so that the token can be updated.

Fundamentally, react hooks requires that this be able to be re-initialized more than once and currently it can't

Reply:
I see. I'm just trying to understand the use-case, are you trying to reinitialize Link with a new token value?

With this PR, you should be able to reinitialize Link if you update the Link onExit callback to something like:

```
onExit: () => {
  const token fetchNewToken();
  // redux action
  updateToken(token);
}
```

Reply:
well, if I'm updating a member, I need to make a call to our backend (which calls plaid) to get the members public token for updating their link. So, on first render, its not available yet. On first render, I make a call to get the token, then I set it as state on the component and thus it rerenders. On the second render, it doesn't update…

Reply:
I'm unable to recreate the `do not call destroy() when Link is open` error in this PR with the storybook example

Reply:
I think I understand now. So when the publicToken is fetched and this component reinitializes, if the user clicks the `open Link` button within 1 second, they will get the `Do not call destroy() while Link is open` error.

I think we can solve this by hooking into the `onExit` callback instead of waiting for an arbitrary 1000ms. Here's the commit to address this: a70bd221bd069d7226f66feb8961b6b53d6fcdd4

Reply:
still getting this 

![image](https://user-images.githubusercontent.com/2091439/77973605-f3021800-72b1-11ea-9b7a-05d2daec0680.png)


All I'm doing is clicking the a "fix" link one time that calls `open`. I'm waiting a while too…